src/Controller/PageController.php

```diff
@@ -115,18 +118,114 @@ public function page(string $slug, Request $request)
             'org_id' => getenv('APP_ENV') === 'prod' ? '00Db0000000egy4' : '00D8E000000E4zz',
             'option_cards' => $optionCardsContent,
+            'slug'               => $slug,
+            'formErrors'         => $formErrors,
+            'formData'           => $formData,
          ]);
     }
 
+    private function sendToSalesforce($params, $formData)
+    {
+
+        if(strpos($params->get('retURL'), 'thank-you-page-newsletters')){
+            $formErrors = $this->validateNewsletterForm($formData);
+        }else{
+            $formErrors = $this->validateForm($formData);
+        }
+        
+        if ($formErrors) {
+            return $formErrors;
+        }else{
+            $response = $this->client->request('POST', getenv('SALESFORCE_WEB_TO_CASE_URL'), [
+                            // these values are automatically encoded before including them in the URL
+                            'query' => $params->all(),
+                        ]);
+
+            if (!is_null($params->get('debug'))) {
+                return new Response(
+                    $response->getContent()
+                );
+            }
+
+            return $this->redirect( $params->get('retURL'));
+        }
+    }
+
+    private function validateNewsletterForm($data){
+
+        $errorMessages = [];
+
+        $errorMessages['nameErr'] = FormValidation::validationName($data['name']);
+        $errorMessages['emailErr'] = FormValidation::validationEmail($data['email']);
+        $errorMessages['companyErr'] = FormValidation::validationCompany($data['company']);
+        
+        foreach ($errorMessages as $type => $value) {
+            if (!empty($errorMessages[$type]['errors'])) {
+                return $errorMessages;
+            }
+        }
+
+        return false;
+    }
+
+    private function validateForm($data){
+
+        $errorMessages = [];
+
+        $errorMessages['nameErr'] = FormValidation::validationName($data['name']);
+        $errorMessages['emailErr'] = FormValidation::validationEmail($data['email']);
+        $errorMessages['phoneErr'] = FormValidation::validationPhone($data['phone']);
+        $errorMessages['companyErr'] = FormValidation::validationCompany($data['company']);
+
+        $errorMessages['aggregationOptionErr'] = FormValidation::validationAggregationOption($data['aggregationOption']);
+
+        $errorMessages['descriptionErr'] = FormValidation::validationDescription($data['description']);
+
+        
+        foreach ($errorMessages as $type => $value) {
+            if (!empty($errorMessages[$type]['errors'])) {
+                return $errorMessages;
+            }
+        }
+
+        return false;
+    }
+
+    private function getFromData($params){
+        return [
+            'name' => $params->get('name'),
```

Comment:
Styling comment:

I would probably have it as 
`'name' => $params->get('name', null),`